packages/ra-tree-ui-materialui/src/TreeShowLayout.js

```diff
@@ -0,0 +1,83 @@
+import React, { cloneElement, Children, Component } from 'react';
+import classNames from 'classnames';
+import { withStyles } from '@material-ui/core/styles';
+
+const CONTAINER_CLASS = 'treenode-content';
+
+const styles = {
+    root: {
+        alignItems: 'center',
+        display: 'flex',
+        flexGrow: 1,
+    },
+};
+
+const sanitizeRestProps = ({
+    cancelDropOnChildren,
+    crudUpdate,
+    dispatchCrudUpdate,
+    getTreeState,
+    isDragging,
+    onSelect,
+    onToggleItem,
+    onUnselectItems,
+    parentSource,
+    startUndoable,
+    translate,
+    undoable,
+    undoableDragDrop,
+    ...rest
+}) => rest;
+
+export class ShowLayout extends Component {
+    handleClick = event => {
+        event.persist();
+        // This ensure clicking on a button does not collapse/expand a node
+        // When clicking on the form (empty spaces around buttons) however, it should
+        // propagate to the parent
+        if (!event.target.matches(`.${CONTAINER_CLASS}`)) {
+            event.stopPropagation();
+        }
+    };
+
+    render() {
+        const {
+            actions,
+            basePath,
+            children,
+            classes,
+            node,
+            resource,
+            ...props
+        } = this.props;
+
+        return (
+            <div
+                className={classNames(CONTAINER_CLASS, classes.root)}
+                onClick={this.handleClick}
+            >
+                {Children.map(
+                    children,
+                    field =>
+                        field
+                            ? cloneElement(field, {
+                                  basePath: field.props.basePath || basePath,
+                                  record: node.record,
+                                  resource,
+                                  ...sanitizeRestProps(props),
```

Comment:
same remark as above, rest props should be passed to the root element, not to the children

Reply:
Actually none of the rest props were needed. I removed this line